src/components/main-view/main-view.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import axios from 'axios';
+import { MovieCard } from '../movie-card/movie-card';
+import { MovieView } from '../movie-view/movie-view';
+
+//Declares the component by extending the React.Component class to inherit all of its lifecyley methods
+export class MainView extends React.Component {
+    constructor() {
+    // Call the superclass constructor
+    // so React can initialize it
+        super();
+
+    // Initialize the state to an empty object so we can destructure it later
+        this.state = {
+            movies: null,
+            selectedMovie: null
+    };
+}
+
+  // One of the "hooks" available in a React Component
+  componentDidMount() {
+    axios.get('https://myflix-movie-application.herokuapp.com/movies')
+      .then(response => {
+        // Assign the result to the state
+          this.setState({
+            //Data from the response/API
+          movies: response.data
+        });
+      })
+      .catch(function (error) {
+        console.log(error);
+      });
+  }
+
+    onMovieClick(movie) {
+        this.setState({
+            selectedMovie: movie
+        });
+    }
+
+  render() {
+    // If the state isn't initialized, this will throw on runtime
+    // before the data is initially loaded
+    const { movies, selectedMovie } = this.state;
+
+    // Before the movies have been loaded
+    if (!movies) return <div className="main-view"/>;
+
+    return (
+     <div className="main-view">
+      {selectedMovie
+         ? <MovieView movie={selectedMovie}/>
```

Comment:
You should pass in an onClick method using `onMovieClick` for your back button.

i.e `<MovieView movie={selectedMovie} goBack={() => this.onMovieClick(undefined)}/>`